interp/constrextern.ml

```diff
@@ -274,17 +274,8 @@ let pattern_printable_in_both_syntax (ind,_ as c) =
 
  (* Better to use extern_glob_constr composed with injection/retraction ?? *)
 let rec extern_cases_pattern_in_scope (scopes:local_scopes) vars pat =
-  (* pboutill: There are letins in pat which is incompatible with notations and
-     not explicit application. *)
-  match pat with
-    | PatCstr(loc,cstrsp,args,na)
-	when !Flags.in_debugger||Inductiveops.constructor_has_local_defs cstrsp ->
-      let c = extern_reference loc Id.Set.empty (ConstructRef cstrsp) in
-      let args = List.map (extern_cases_pattern_in_scope scopes vars) args in
-      CPatCstr (loc, c, Some (add_patt_for_params (fst cstrsp) args), [])
-    | _ ->
   try
-    if !Flags.raw_print || !print_no_symbol then raise No_match;
+    if !Flags.in_debugger || !Flags.raw_print || !print_no_symbol then raise No_match;
```

Comment:
> For instance, it could be really hard to debug a printing time exception (say one due to threading the wrong environment to the printer), don't you think?

Yes, I had a lot of pain many times in the past chasing printing time exception in the debugger (see e.g. 5cd253c, 287d6f8, 65e645a5f, 3be4b91c09, ...). I eventually developed some methodology to deal with that but it is still a few minutes to spot problems (by brute force, adding ad hoc handlers at all suspicious places).

By the way, we can thank ocaml (and Xavier Clerc). It used to be a time where an exception in the debugger was simply killing the debugger...